src/robotFactory.js

```diff
@@ -1,10 +1,92 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
```

Comment:
som if you change next comment - it will be extra
```suggestion
  constructor(name, weight, coords, chipVersion) {
```